platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -326,37 +325,16 @@ def install_firmware(self, image_path):
             return False
 
         cmdline = self.CPLD_UPDATE_COMMAND.format(mst_dev_list[0], image_path)
-        outputline = ""
         success_flag = False
-        try:
-            proc = subprocess.Popen(cmdline, stdout=subprocess.PIPE, shell=True, stderr=subprocess.STDOUT)
-            while True:
-                out = proc.stdout.read(1)
-
-                if out == '' and proc.poll() != None:
-                    break
-
-                if out != '':
-                    sys.stdout.write(out)
-                    sys.stdout.flush()
-                    outputline += out
-
-                if (out == '\n' or out == '\r') and len(outputline):
-                    m = re.search(self.CPLD_INSTALL_SUCCESS_FLAG, outputline)
-                    if m and m.group(0) == self.CPLD_INSTALL_SUCCESS_FLAG:
-                        success_flag = True
 
-            if proc.returncode:
-                print("ERROR: Upgrade CPLD failed, return code {}".format(proc.returncode))
-                success_flag = False
-
-        except OSError as e:
-            raise RuntimeError("Failed to execute command {} due to {}".format(cmdline, repr(e)))
+        try:
+            subprocess.check_call(cmdline, stderr=subprocess.STDOUT, shell=True)
+            success_flag = True
+        except subprocess.CalledProcessError as e:
+            print("ERROR: Failed to upgrade CPLD: rc={}".format(e.returncode))
 
         if success_flag:
             print("INFO: Refresh or power cycle is required to finish CPLD installation")
-        else:
```

Comment:
Why "else" is removed? I see ```success_flag``` is still in use.
Another question, if you removed whole ```Popen``` logic and now ```check_call``` + ```try``` are used. Do you still need this ```success_flag```?

Reply:
@volodymyrsamotiy the `success_flag` is as indicator for upper layer that update was successful. We need to keep it as it is by design. As for the current implementation - we need it to print the user hint only in case update was successful. If the error happens - only the relevant message will be printed. This helps to avoid multiple duplicated error messages.

Reply:
By calling `subprocess.check_call`, is it able to print the output of subcommand to screen in realtime?

Reply:
@stephenxs yes, it is.